__tests__/wallet-precalculation.test.js

```diff
@@ -0,0 +1,191 @@
+import fs from 'fs';
+import { WALLET_CONSTANTS } from './integration/configuration/test-constants';
+import {
+  multisigWalletsData, WalletPrecalculationHelper,
+} from '../src/helpers/wallet-precalculation.helper';
+import precalculatedMultisig from './integration/configuration/precalculated-multisig-wallets.json';
+
+jest.mock('fs', () => ({
+  promises: {
+    readFile: jest.fn(),
+    writeFile: jest.fn()
+  }
+}));
+
+const fakeWallet = {
+  isUsed: false,
+  words: 'waste edit random program emotion vapor people scheme axis phone peasant end real tired panther build subject pyramid ceiling glory nature proud monitor empower',
+  addresses: ['WYLW8ujPemSuLJwbeNvvH6y7nakaJ6cEwT',
+    'WUfmqHWQZWn7aodAFadwmSDfh2QaUUgCRJ',
+    'Wi8zvxdXHjaUVAoCJf52t3WovTZYcU9aX6',
+    'WSu4PZVu6cvi3aejtG8w7bomVmg77DtqYt',
+    'WVbN5f2vS4TyP2XmS2x8aG2Qc5oACy3e7L',
+    'WY6wbpDWE1EFHpKzaknp9hMSrhZiH5jJ9s',
+    'WXQs68APxW9VTpbjK9FBMWrwWidxTwZpoN',
+    'WeWod5pie133e6DW3PRjhsXDp4Yq6d2D9a',
+    'Waf7MEhFy6ibcrA4sCKuKidEcW9p7E7C48',
+    'WXzfjqxKqe44yLfrsx3bmbXFRehhEgUofC',
+    'WfWFSZYyAaGTEDuLyMPAdotLuiMhkDUkyS',
+    'Wg7BaqXikZBDQcWipgauMSR38qAvsUVZvy',
+    'WhFKDjgBRScAqZHfc1XvWHj3nwVAgUvuUf',
+    'Wjos8haw4WnqkK2KauC19Xb53XLFzx5bB5',
+    'Wf5ivZqY6Ky7rp1MTPvWn2TgXgykYMos6x',
+    'WQ3YUhdLtL3D1RZvj1wGz2BSGW1bM9Wj12',
+    'WXfhSWZYaEQbkWAjFJF7bBjYtv8jHavYC6',
+    'WYd8CMafF44mnVZUBZrdPqcNFFVhCm737s',
+    'Wg7wUVBBuAKenAa6FWBi9pJAE83rHmvSzQ',
+    'WPm5JsTnYyuWQfpwRStDRshJb7X4XXPih3',
+    'Wgop7ug1pTDJhhUVMzAUTEznzRRLEdzAGr',
+    'WcRom2aEeLkcQMmiv1myTLZRG1Q6G6htEJ'
+  ],
+};
+
+describe('wallet addresses precalculation', () => {
+  it('correctly generates addresses for a known common wallet', () => {
+    const { words } = WALLET_CONSTANTS.genesis;
+    const generatedWallet = WalletPrecalculationHelper.generateWallet({ words });
+
+    for (const i in generatedWallet.addresses) {
+      expect(generatedWallet.addresses[i]).toBe(WALLET_CONSTANTS.genesis.addresses[i]);
+    }
+  });
+
+  it('correctly generates addresses for a new wallet', () => {
+    const generatedWallet = WalletPrecalculationHelper.generateWallet();
+
+    expect(generatedWallet).toHaveProperty('words');
+    expect(generatedWallet.words.split(' ')).toHaveProperty('length', 24);
+    expect(generatedWallet.addresses).toHaveProperty('length', 22);
+  });
+
+  it('correctly generates multiple new wallets with a specified amount', () => {
+    jest.spyOn(WalletPrecalculationHelper, 'generateWallet')
```

Comment:
When the jest infrastructure is up, address generation inside it takes a lot longer than usual. That is, actually, the behaviour we've seen on the integration tests before.

Since the `generateWallet` function itself is already being tested elsewhere on this suite, I only wanted to assure it was called multiple times here to compose the results. No need to wait over a minute of busy CPU time calculating those addresses that won't affect the test assertions.